spring-cloud-dataflow-core/src/main/java/org/springframework/cloud/dataflow/core/StreamDefinition.java

```diff
@@ -127,25 +127,21 @@ public String toString() {
 
 
 	/**
-	 * Iterator that prevents mutation of its backing data structure.
-	 *
-	 * @param <T> the type of elements returned by this iterator
+	 * Iterator that walks through ModuleDefinitions in deployment order, that is, from downstream (sink, if present)
+	 * to upstream (source) module.
+	 * Also prevents mutation of its backing data structure.
 	 */
-	private static class ReadOnlyIterator<T> implements Iterator<T> {
-		private final Iterator<T> wrapped;
-
-		public ReadOnlyIterator(Iterator<T> wrapped) {
-			this.wrapped = wrapped;
-		}
+	public class DeploymentOrderIterator implements Iterator<ModuleDefinition> {
```

Comment:
Before these changes, the iterator instance actually delegated to a wrapped iterator, and that way the state was consistent with that wrapped iterator, even if there were more than one instance of the `ReadOnlyIterator`. With these changes, there could be more than one instance of this iterator, and each would have its own state for the position. Is that something we want to allow?